src/suggestion.cpp

```diff
@@ -0,0 +1,325 @@
+
+    xquery = Xapian::Query(Xapian::Query::OP_OR, xquery, subquery_phrase);
+    xquery = Xapian::Query(Xapian::Query::OP_OR, xquery, subquery_anchored);
+  }
+
+  return xquery;
+}
+
+
+SuggestionSearcher::SuggestionSearcher(const Archive& archive) :
+    mp_internalDb(nullptr),
+    m_archive(archive)
+{}
+
+SuggestionSearcher::SuggestionSearcher(const SuggestionSearcher& other) = default;
+SuggestionSearcher& SuggestionSearcher::operator=(const SuggestionSearcher& other) = default;
+SuggestionSearcher::SuggestionSearcher(SuggestionSearcher&& other) = default;
+SuggestionSearcher& SuggestionSearcher::operator=(SuggestionSearcher&& other) = default;
+SuggestionSearcher::~SuggestionSearcher() = default;
+
+SuggestionSearch SuggestionSearcher::suggest(const SuggestionQuery& query)
+{
+  if (!mp_internalDb) {
+    initDatabase();
+  }
+  return SuggestionSearch(mp_internalDb, query);
+}
+
+void SuggestionSearcher::setVerbose(bool verbose)
+{
+  if (!mp_internalDb) {
+    initDatabase();
+  }
+  mp_internalDb->m_verbose = verbose;
+}
+
+void SuggestionSearcher::initDatabase()
+{
+    mp_internalDb = std::make_shared<SuggestionDataBase>(m_archive);
+}
+
+SuggestionSearch::SuggestionSearch(std::shared_ptr<SuggestionDataBase> p_internalDb, const SuggestionQuery& query)
+ : mp_internalDb(p_internalDb),
+   mp_enquire(nullptr),
+   m_query(query)
+{
+}
+
+SuggestionSearch::SuggestionSearch(SuggestionSearch&& s) = default;
+SuggestionSearch& SuggestionSearch::operator=(SuggestionSearch&& s) = default;
+SuggestionSearch::~SuggestionSearch() = default;
+
+SuggestionQuery& SuggestionQuery::setQuery(const std::string& query) {
+    m_query = query;
+    return *this;
+}
+
+int SuggestionSearch::getEstimatedMatches() const
+{
+  if (mp_internalDb->hasDatabase()) {
+    try {
+      auto enquire = getEnquire();
+      // Force xapian to check at least one document even if we ask for an empty mset.
+      // Else, the get_matches_estimated may be wrong and return 0 even if we have results.
+      auto mset = enquire.get_mset(0, 0, 1);
+      return mset.get_matches_estimated();
+    } catch(...) {
+      std::cerr << "Query Parsing failed, Switching to search without index." << std::endl;
+    }
+  }
+
+  return mp_internalDb->m_archive.findByTitle(m_query.m_query).size();
+}
+
+const SuggestionResultSet SuggestionSearch::getResults(int start, int maxResults) const {
+    if (mp_internalDb->hasDatabase())
+    {
+      try {
+        auto enquire = getEnquire();
+        auto mset = enquire.get_mset(start, maxResults);
+        return SuggestionResultSet(mp_internalDb, std::move(mset));
+      } catch(...) {
+        std::cerr << "Query Parsing failed, Switching to search without index." << std::endl;
+      }
+    }
+    return SuggestionResultSet(mp_internalDb->m_archive.findByTitle(m_query.m_query));
```

Comment:
This doesn't respect start and maxResults.
It should be "easy" to create a new EntryRange with `m_begin = m_begin + start` and `m_end = m_end + start + maxResult` (modulo checking that range limits are valid)

Reply:
@mgautierfr Should we add this feature into the archive API itself like:
`EntryRange<EntryOrder::titleOrder> findByTitle(std::string title, int start, int maxResults)` 
What do you think?